lib/dms.js

```diff
@@ -46,9 +46,47 @@ class DmsModel {
     await Promise.all(datapackage.resources.map(async resource => {
       resource.views = await this.getResourceViews(resource.id)
     }))
+
+    // Fetch dictionary fields for the resources
+    await Promise.all(datapackage.resources.map(async resource => {
+      //TODO Check resource type and make
+      resource.schema = await this.getResourceSchema(resource.id)
+    }))
     return datapackage
   }
 
+  async getResourceSchema(resourceId) {
+    try{
+      const action = 'datastore_search'
+      let url = new URL(resolve(this.api, action))
+      let response = await fetch(url, {
+        method: 'POST',
+        headers: { 
+          'Content-Type': 'application/x-www-form-urlencoded'
+        },
+        body: JSON.stringify({
+          'resource_id' : resourceId
+       }) 
+      })
+
+      if (response.status !== 200) {
+        throw response
+      }
+      response = await response.json()
+      const { fields } = response.result
+      for (let i = 0; i < fields.length; i++) {
+        fields[i] = utils.ckanSchemaToTableSchema(fields[i])
+      }
+      delete response.result.fields
+      return {
+        fields: fields
+      }
+    } catch (e) {
+      console.warn('Error fetching resource fields', e)
```

Comment:
Let's make it clear what is the error:

`Failed fetching datastore api when trying to fetch data dictionary`